client/app/components/QueryEditor.jsx

```diff
@@ -260,6 +260,17 @@ class QueryEditor extends React.Component {
 
     const isExecuteDisabled = this.props.queryExecuting || !this.props.canExecuteQuery;
 
+    function executeCheck() {
+      if (this.props.dataSource.dry_run && !this.props.dryRunLoading &&
+        this.props.dryRunResult != null && this.props.dryRunResult.size > 1e12) {
+        if (!confirm('You are about to submit a query that reads over 1tb of data, are you sure?')) {
```

Comment:
replace text here with:"You are about to submit a query that will process over 1TB of data, this may take a while (and cost a lot of money), please contact your Devision BI person before you continue."